Model/Post.php

```diff
@@ -10,209 +10,252 @@
+     *
+     * @param string $bodyPreview
+     * @return Post
+     */
+    public function setBodyPreview($bodyPreview)
+    {
+        $this->bodyPreview = $bodyPreview;
+
+        return $this;
     }
 
+    /**
+     * Get body (the post's content)
+     *
+     * @return string
+     */
     public function getBody()
     {
         return $this->body;
     }
 
+    /**
+     * Set body (the post's content)
+     *
+     * @param string $body
+     * @return Post
+     */
     public function setBody($body)
     {
         $this->body = $body;
+
+        return $this;
     }
 
+    /**
+     * Get publication date
+     *
+     * @return \DateTime
+     */
     public function getDate()
     {
         return $this->date;
     }
 
+    /**
+     * Set publication date
+     *
+     * @param \DateTime $date
+     * @return Post
+     */
     public function setDate($date)
     {
         $this->date = $date;
-    }
-
-    public function getParent()
-    {
-        return $this->getBlog();
-    }
-
-    public function getBlog()
-    {
-        return $this->blog;
-    }
-
-    public function setBlog(Blog $blog)
-    {
-        $this->blog = $blog;
 
-        // The user can create a post from Admin, so
-        // we let them choose and automatically make
-        // this Post a child of selected blog.
-        $this->parent = $blog;
-    }
-
-    public function getBodyPreview($length = 255)
-    {
-        $suffix = strlen($this->body) > $length ? ' ...' : '';
-
-        return substr($this->body, 0, 255).$suffix;
-    }
-
-    public function getRoutes()
-    {
-        return $this->routes;
+        return $this;
     }
 
+    /**
+     * Is publishable
+     *
+     * @return bool
+     */
     public function isPublishable()
     {
         return $this->isPublishable;
     }
 
+    /**
+     * Set publishable
+     *
+     * @param bool $publishable
+     * @return Post
+     */
     public function setPublishable($publishable)
     {
         $this->isPublishable = $publishable;
+
+        return $this;
     }
 
+    /**
+     * Get publish start date
+     *
+     * @return \DateTime
+     */
     public function getPublishStartDate()
     {
         return $this->publishStartDate;
     }
 
+    /**
+     * Set publish start date
+     *
+     * @param \DateTime $publishStartDate
+     * @return Post
+     */
     public function setPublishStartDate(\DateTime $publishStartDate = null)
     {
         $this->publishStartDate = $publishStartDate;
+
+        return $this;
     }
 
+    /**
+     * Get publish end date
+     *
+     * @return \DateTime
+     */
     public function getPublishEndDate()
     {
         return $this->publishEndDate;
     }
 
+    /**
+     * Set publish end date
+     *
+     * @param \DateTime $publishEndDate
+     * @return Post
+     */
     public function setPublishEndDate(\DateTime $publishEndDate = null)
     {
         $this->publishEndDate = $publishEndDate;
+
+        return $this;
```

Comment:
We don't use fluid interfaces in other places in the CMF, not sure we want to start here.